gcloud/connection.py

```diff
@@ -94,6 +93,25 @@ def http(self):
                 self._http = self._credentials.authorize(self._http)
         return self._http
 
+    @staticmethod
+    def _scoped_credentials(credentials, scope):
+        """Create a scoped set of credentials if it is required.
+
+        :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                           :class:`NoneType`
+        :param credentials: The OAuth2 Credentials to add a scope to.
+
+        :type scope: list of URLs
+        :param scope: the effective service auth scopes for the connection.
+
+        :rtype: :class:`oauth2client.client.OAuth2Credentials` or
+                :class:`NoneType`
+        :returns: A new credentials object that has a scope added (if needed).
+        """
+        if credentials and credentials.create_scoped_required():
```

Comment:
Well as of right now in the `Connection` constructors we call

```python
credentials = self._scoped_credentials(credentials, SCOPE)
```

unconditionally. But if the user has already added a scope to the credentials, we can leave it as is.


Would you like a check for the existing scope and `SCOPE` to agree?